### ISSUE TITLE
GradientDescent behavior is stochastic even when using random_state

### ISSUE THREAD

**Issue by omiguelpinheiro** [omiguelpinheiro/Prediction-Models#1]
Linear Regression gives different results for same set of parameters, when using any GradientDescent algorithm

**Comment by omiguelpinheiro** [comment]
Using LinearRegression.solve() also gives different results every execution. The problem is probably related to the train_test_split() method.

**Comment by omiguelpinheiro** [comment]
The random number generetor (RNG) on preprocess.train_test_split() used a seed independent from the np.random.seed(). Just moved the random state to the RNG arguments and the behavior stopped.

Commit:  bdd91ae81350135b6286d3f7783911c2d92cb855

**Comment by omiguelpinheiro** [comment]
The random number generetor (RNG) on preprocess.train_test_split() used a seed independent from the np.random.seed(). Just moved the random state to the RNG arguments and the behavior stopped.

Commit:  bdd91ae81350135b6286d3f7783911c2d92cb855

**Comment by omiguelpinheiro** [comment]
In train_test_split() the random number generator uses a private seed to randomize it's outputs, just moved the random_state provided in the argument to the random number generator parameters and the behavior is now deterministic.

Commit: bdd91ae81350135b6286d3f7783911c2d92cb855
